src/main/java/edu/harvard/iq/dataverse/settings/SettingsServiceBean.java

```diff
@@ -420,7 +410,26 @@ Whether Harvesting (OAI) service is enabled
          * Indicates if other terms of use are active or not.
          */
         AllRightsReservedTermsOfUseActive,
-        RestrictedAccessTermsOfUseActive;
+        RestrictedAccessTermsOfUseActive,
+
+        /**
+         * Size limits for generating thumbnails on the fly
+         * (i.e., we'll attempt to generate a thumbnail on the fly if the
+         * size of the file is less than this)
+         */
+        ThumbnailImageSizeLimit,
+        ThumbnailPDFSizeLimit,
```

Comment:
I think that this properties are currently used as system properties and it will be quite easy to replace:
Take a look at: SystemConfig.getThumbnailSizeLimitImage and SystemConfig.getThumbnailSizeLimitPDF

Reply:
Unfortunately it is also duplicated in another place where it is static ImageThumbConverter:965
`private static long getThumbnailSizeLimit(String type) {
        String option = null;
        if ("Image".equals(type)) {
            option = System.getProperty("dataverse.dataAccess.thumbnail.image.limit");
        } else if ("PDF".equals(type)) {
            option = System.getProperty("dataverse.dataAccess.thumbnail.pdf.limit");
        }`

Reply:
ok, then lets leave it for now